Source/ACE.Database/DatabaseManager.cs

```diff
@@ -69,5 +72,18 @@ public static void Stop()
             if (serializedShardDb != null)
                 serializedShardDb.Stop();
         }
+
+        private static readonly Dictionary<string, ServerVersion> cachedServerVersions = new();
+
+        public static ServerVersion CachedServerVersionAutoDetect(string database, string connectionString)
+        {
+            if (cachedServerVersions.TryGetValue(database, out ServerVersion serverVersion)) { return serverVersion; }
```

Comment:
is there any chance of this function being called concurrently? as currently written, not thread-safe